egg/core/language_analysis.py

```diff
@@ -17,6 +17,31 @@
 from .interaction import Interaction
 
 
+def gap_mi_first_second(attributes, representations):
+    gaps = torch.zeros(representations.size(1))
+    non_constant_positions = 0.0
+
+    for j in range(representations.size(1)):
+        symbol_mi = []
+        h_j = None
+        for i in range(attributes.size(1)):
+            x, y = attributes[:, i], representations[:, j]
+            info = mutual_info(x, y)
+            symbol_mi.append(info)
+
+            if h_j is None:
+                h_j = calc_entropy(y)
+
+        symbol_mi.sort(reverse=True)
```

Comment:
minorest nit ever: could be a heap

Reply:
For efficiency reasons? Is it really worth it? I'll give it a try

Reply:
not really worth it :)